nixos/modules/services/networking/sabnzbd/default.nix

```
@@ -0,0 +1,609 @@
+                    };
+                    ssl_verify = mkOption {
+                      type =
+                        types.coercedTo
+                          (types.enum [
+                            "strict"
+                            "allow injection"
+                            "none"
+                          ])
+                          (
+                            val:
+                            if val == "strict" then
+                              3
+                            else if val == "allow injection" then
+                              2
+                            else
+                              0
+                          )
+                          (
+                            types.enum [
+                              0
+                              1
+                              2
+                              3
+                            ]
+                          );
+
+                      description = ''
+                        Level of TLS verification. Supported values:
+                        3, 'strict'          -- strict (normal) verification
+                        2, 'allow injection' -- allow locally injected certificates
+                        1, 0, 'none'         -- no verification
+                      '';
+                      default = "strict";
+                    };
+                    expire_date = mkOption {
+                      type = types.nullOr types.str;
+                      description = ''
+                        If Notifications are enabled and an expiry date is
+                        set, warn 5 days before expiry. This setting
+                        does not automatically disable the server.
+                        Expected format: yyyy-mm-dd
+                      '';
+                      default = null;
+                    };
+                  };
+                }
+              );
+            };
+          };
+        };
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    warnings =
+      if cfg.configFile != null then
+        [
+          ''
+            `sabnzbd.configFile` is deprecated, consider using `sabnzbd.settings` instead.
+            If you have values set in `sabnzbd.settings` set, they will be ignored.
+          ''
+        ]
+      else
+        [ ];
+
+    users.users = mkIf (cfg.user == "sabnzbd") {
+      sabnzbd = {
+        uid = config.ids.uids.sabnzbd;
+        group = cfg.group;
+        description = "sabnzbd user";
+      };
+    };
+
+    users.groups = mkIf (cfg.group == "sabnzbd") {
+      sabnzbd.gid = config.ids.gids.sabnzbd;
+    };
+
+    systemd.services.sabnzbd =
+      let
+        secretIniQuoted = lib.strings.optionalString (cfg.secretsFile != null) "'${cfg.secretsFile}'";
+        optionalWriteableConfigInclusion = lib.strings.optionalString cfg.allowConfigWrite ''[ -e '${totalIni}' ] && files=('${totalIni}' "''${files[@]}")'';
```

```suggestion
        secretIniQuoted = optionalString (cfg.secretsFile != null) "'${cfg.secretsFile}'";
        optionalWriteableConfigInclusion = optionalString cfg.allowConfigWrite ''[ -e '${totalIni}' ] && files=('${totalIni}' "''${files[@]}")'';
```
Also you can safely take optionalString from lib directly.